src/Generator/Passes/SpecializationMethodsWithDependentPointersPass.cs

```diff
@@ -120,6 +120,7 @@ private static Method GetExtensionMethodForDependentPointer(Method specializedMe
                         new QualifiedType(new TagType(specializedMethod.Namespace))));
                     thisParameter.Name = "@this";
                     thisParameter.Kind = ParameterKind.Extension;
+                    thisParameter.Namespace = specializedMethod;
```

Comment:
I said: the name-space must be the function the parameter belongs to. You set the name-space to one and then add to another.